tools/check_release.py

```diff
@@ -0,0 +1,455 @@
+    with open(bak_file, 'r') as ip, open(lib_file, 'w') as op:
+        for line in ip:
+
+            opline = line
+            
+            regexp = lib_re.match(line)
+            if regexp:
+                opline = 'https://developer.mbed.org/users/' + user + '/code/' + library + '/#' + ref
+                updated = True
+    
+            op.write(opline)
+
+    if updated:
+
+        # Setup the default commit message
+        commit_message = '"Updating ' + library + ' to ' + ref + '"' 
+ 
+        # Setup and run the commit command. Need to use the rawcommand in the hglib for this in order to pass
+        # the string value to the -m option. run_cmd using subprocess does not like this syntax.
+        try:
+            client.rawcommand(['commit','-m '+commit_message, lib_file])
+
+            cmd = ['hg', 'push', '-f', repo]
+            run_cmd(cmd, exit_on_failure=True)
+            
+        except:
+            print("Lib file already up to date and thus nothing to commit")
+                            
+    os.chdir(cwd)
+    return updated
+
+def get_sha(repo_path, library):
+    """ Gets the latest SHA for the library specified. The library is assumed to be
+        located at the repo_path. If a SHA cannot be obtained this script will exit.
+
+    Args:
+    library - library name
+    repo_path - path to the user's repo area
+    
+    Returns:
+    sha - last commit SHA
+    """
+    cwd = os.getcwd()
+    sha = None
+    os.chdir(abspath(repo_path + '/' + library))
+    
+    cmd = ['hg', 'log', '-l', '1']
+    ret, output = run_cmd_with_output(cmd, exit_on_failure=True)
+    
+    # Output should contain a 4 line string of the form:
+    # changeset:   135:176b8275d35d
+    # tag:         tip
+    # user:        <>
+    # date:        Thu Feb 02 16:02:30 2017 +0000
+    # summary:     Release 135 of the mbed library
+    # All we want is the changeset string after version number
+    
+    lines = output.split('\n')
+    fields = lines[0].split(':')
+    sha = fields[2]
+    
+    os.chdir(cwd)
+    return sha
+
+def get_latest_library_versions(repo_path):
+    """ Returns the latest library versions (SHAs) for 'mbed' and 'mbed-dev'. 
+    If the SHAs cannot be obtained this script will exit.
+
+    Args:
+    repo_path - path to the user's repo area
+    
+    Returns:
+    mbed - last commit SHA for mbed library
+    mbed_dev - last commit SHA for mbed_dev library
+    
+    """
+
+    mbed = get_sha(repo_path, 'mbed')
+    mbed_dev = get_sha(repo_path, 'mbed-dev')
+
+    return mbed, mbed_dev
+
+if __name__ == '__main__':
+    
+    # Read configuration data
+    json_data = json.load(open(os.path.join(os.path.dirname(__file__), "check_release.json")))
+    
+    # Debug output off by default
+    dbg = False
+
+    supported_targets = []
+    
+    # Get a list of the officially supported mbed-os 2 targets
+    for tgt in OFFICIAL_MBED_LIBRARY_BUILD:
+        if 'ARM' in tgt[1]:
```

Comment:
I don't think is a necessary check as the tools already do this for you when you call `get_mbed_official_release('2')`. And in fact, it's incorrect because it excludes the one mbed 2 target that only supports uARM: https://github.com/ARMmbed/mbed-os/blob/master/targets/targets.json#L989